build.zig

```diff
@@ -140,6 +140,7 @@ pub fn build(b: *std.build.Builder) !void {
     const gen_exe = b.addExecutable(.{
         .name = "zls_gen",
         .root_source_file = .{ .path = "src/config_gen/config_gen.zig" },
+        .target = target,
```

Comment:
This breaks cross compilation because the `zls_gen` executable need to always target the host machine.
```suggestion
        .target = b.host,
```